src/main/java/seedu/address/logic/commands/CloneCommand.java

```diff
@@ -104,6 +136,167 @@ public boolean equals(Object other) {
 
         // state check
         CloneCommand e = (CloneCommand) other;
-        return index.equals(e.index);
+        return index.equals(e.index)
+                && clonePersonDescriptor.equals(e.clonePersonDescriptor);
+    }
+
+    /**
+     * Stores the details to clone the person with. Each non-empty field value will replace the
+     * corresponding field value of the person.
+     */
+    public static class ClonePersonDescriptor {
+        private Name name;
+        private Phone phone;
+        private Email email;
+        private Address address;
+        private Gender gender;
+        private Birthdate birthdate;
+        private Race race;
+        private Religion religion;
+        private Set<Survey> surveys;
+        private Set<Tag> tags;
+
+        public ClonePersonDescriptor() {}
+
+        /**
+         * Copy constructor.
+         * A defensive copy of {@code tags} is used internally.
+         */
+        public ClonePersonDescriptor(CloneCommand.ClonePersonDescriptor toCopy) {
+            setName(toCopy.name);
+            setPhone(toCopy.phone);
+            setEmail(toCopy.email);
+            setAddress(toCopy.address);
+            setGender(toCopy.gender);
+            setBirthdate(toCopy.birthdate);
+            setRace(toCopy.race);
+            setReligion(toCopy.religion);
+            setSurveys(toCopy.surveys);
+            setTags(toCopy.tags);
+        }
+
+        /**
+         * Returns true if at least one field is edited.
+         */
+        public boolean isAnyFieldEdited() {
+            return CollectionUtil.isAnyNonNull(name, phone, email, address, gender, birthdate,
+                    race, religion, surveys, tags);
+        }
+
+        public void setName(Name name) {
```

Comment:
Might want to add JavaDocs since these are public methods.

Reply:
Actually realized that they did not include it in the code from the start, you can just disregard this.